index.js

```diff
@@ -0,0 +1,75 @@
+const https = require('https');
+const { parseString } = require('xml2js');
+
+async function getAllUrlsFromSitemapIndex(siteUrl) {
+  try {
+    const sitemapIndexUrl = `${siteUrl}/sitemap.xml`;
+    const sitemapIndexResponse = await fetchContent(sitemapIndexUrl);
+
+    const sitemapIndexXml = sitemapIndexResponse.data;
+    const sitemapIndexResult = await parseXml(sitemapIndexXml);
+    const sitemapUrls = sitemapIndexResult.sitemapindex.sitemap.map(sitemap => sitemap.loc[0]);
+    const fetchSitemapPromises = sitemapUrls.map(url => fetchContent(url));
+    const sitemapResponses = await Promise.all(fetchSitemapPromises);
+
+    let allUrls = [];
+    for (const response of sitemapResponses) {
+      const sitemapResult = await parseXml(response.data);
+
+      if (sitemapResult.urlset && sitemapResult.urlset.url) {
+        const sitemapUrls = sitemapResult.urlset.url.map(urlObj => urlObj.loc[0]);
+        allUrls = allUrls.concat(sitemapUrls);
```

Comment:
I think we can use push() here. 

Please check blog for more details : https://dev.to/uilicious/javascript-array-push-is-945x-faster-than-array-concat-1oki